### PR TITLE
[4.2] Move check for discoverable extensions to model

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Ongoing effort to inject the database into components and get rid of the deprecated Factory::getDbo(); calls. This pr moves a database function in the discover view to the model where the database is available.

### Testing Instructions
Go to the site /administrator/index.php?option=com_installer&view=discover.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### CODE REVIEW

File: administrator/components/com_installer/src/View/Discover/HtmlView.php [hunk 1]
```diff
@@ -92,26 +92,4 @@ protected function addToolbar()
...
-	 *
-	 * Checks uninstalled extensions in extensions table.
-	 *
-	 * @return  boolean  True if there are discovered extensions on the database.
-	 *
-	 * @since   3.5
-	 */
-	public function checkExtensions()
-	{
-		$db = Factory::getDbo();
```
Comment (Quy) [inline administrator/components/com_installer/src/View/Discover/HtmlView.php:107]: Please delete: https://github.com/joomla/joomla-cms/blob/6852b2c88a8bf9b2a2cb7d89c68c865d85fbd84e/administrator/components/com_installer/src/View/Discover/HtmlView.php#L14

File: administrator/components/com_installer/src/Model/DiscoverModel.php [hunk 2]
```diff
@@ -305,4 +305,24 @@ protected function getEmptyStateQuery()
 
 		return $query;
 	}
+
+	/**
+	 * Checks uninstalled extensions in extensions table.
```
Comment (brianteeman) [inline administrator/components/com_installer/src/Model/DiscoverModel.php]:
```suggestion
	 * Checks for not installed extensions in extensions table.
```

i appreeciate this was wrong before your pr

File: administrator/components/com_installer/src/Model/DiscoverModel.php [hunk 3]
```diff
@@ -305,4 +305,24 @@ protected function getEmptyStateQuery()
...
+	 *
+	 * @return  boolean  True if there are discovered extensions on the database.
```
Comment (brianteeman) [inline administrator/components/com_installer/src/Model/DiscoverModel.php]:
```suggestion
	 * @return  boolean  True if there are discovered extensions in the database.
```

i appreeciate this was wrong before your pr

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 469dbf7302266fcc782fceebe9fe738a4633924a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37605">issues.joomla.org/tracker/joomla-cms/37605</a>.</sub>

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on a9af3c7ff853bbeea44e2e435918803a232a2083<br /><br />Tested successfully in Joomla 4.2.0-alpha2 of 25 April in Wampserver 3.2.8 using PHP 8.0.15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37605">issues.joomla.org/tracker/joomla-cms/37605</a>.</sub>

**Comment by SumCompanyInc** [comment]
I have tested this item :white_check_mark: successfully on a9af3c7ff853bbeea44e2e435918803a232a2083<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37605">issues.joomla.org/tracker/joomla-cms/37605</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37605">issues.joomla.org/tracker/joomla-cms/37605</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody

**Pull request merged**